src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,184 @@
+        if (head == null) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Element doesn't exist at " + index);
+        }
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        return getNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndex(index);
+        Node<T> oldNode = getNode(index);
+        T oldValue = oldNode.element;
+        oldNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index);
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = head; x != null; x = x.next) {
+            if (x.element == object) {
+                unlink(x);
+                return true;
+            }
+        }
+        if (object != null) {
+            for (Node<T> x = head; x != null; x = x.next) {
+                if (object.equals(x.element)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkFirst(T value) {
+        Node<T> head = this.head;
+        Node<T> newNode = new Node<>(null, value, head);
+        this.head = newNode;
+        if (head == null) {
+            tail = newNode;
+        } else {
+            head.prev = newNode;
+        }
+        size++;
+    }
+
+    private void linkLast(T value) {
+        Node<T> tail = this.tail;
+        Node<T> newNode = new Node<>(tail, value, null);
+        this.tail = newNode;
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T value, Node<T> node) {
+        Node<T> prev = node.prev;
+        Node<T> newNode = new Node<>(prev, value, node);
+        node.prev = newNode;
+        if (prev == null) {
+            head = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> node;
+        int i;
+        if (size == 0) {
+            node = null;
+        }
```

Comment:
No need to check it, when your size `0` you throw an exception in `checkPositionIndex`